src/Service/Modules/VatCodeModule.cs

```diff
@@ -44,13 +53,19 @@ private object GetVatCodes()
                 .WithView("Index");
         }
 
-        private object UpdateSernosConfig(string code)
+        private object UpdateVatCode(string code)
         {
             var resource = this.Bind<VatCodeResource>();
-
-            var result = this.vatCodeService.Update(code, resource);
+            try
+            {
+                var result = this.vatCodeService.Update(code, resource);
             return this.Negotiate.WithModel(result).WithMediaRangeModel("text/html", ApplicationSettings.Get)
                 .WithView("Index");
-        }
+            }
+            catch (DbUpdateException e)
+            {
+                return new BadRequestResult<VatCode>(e.InnerException.Message);
```

Comment:
You'll want to put in a meaningful message for the user here, and log the actual exception too.. otherwise that information will be lost

Reply:
I think the Nancy [onError ](https://github.com/NancyFx/Nancy/wiki/The-Application-Before%2C-After-and-OnError-pipelines#intercepting-the-request-when-an-error-occurred )hook we set up in the bootstrapper catches and logs any exceptions:  so hopefully no danger of anything being swallowed.  As for the message being logged, this will just be the Oracle exception which I agree isn't super user friendly. It does give details about any mistake they have made though (which could be useful when they inevitably phone IT support to report errors like this).

Reply:
Cool, didn't realise about the onError hook.. nice.

Reply:
For the message displayed to the user I'd put in something specific enough that we'd understand the source of the exception but simple enough that they can tell us without being confused and having to copy and paste some stack trace.. I'm guessing we'd still need to go and look at the event log to unpick what happened? but yeh.. guess it is a bit of a trade off if that message made it obvious to us what the cause is..